src/bin/index.js

```diff
@@ -1,20 +1,24 @@
 #!/usr/bin/env node
 
 const { error } = require('../lib/log');
-require('../cli/check-node-version');
+const checkNodeVersion = require('../cli/check-node-version');
 
-const main = require('../lib/main');
+try {
+  checkNodeVersion();
+} catch (e) {
+  error(e.message);
+  process.exitCode = 1;
+}
 
+const main = require('../lib/main');
 (async () => {
   let returnCode;
   try {
     returnCode = await main(process.argv, process.env);
-  }
-  catch (e) {
-    error(e);
+  } catch (e) {
+    error(e.message);
```

Comment:
Maybe the stack trace would be useful here, if it's not an error that we expect.

Reply:
This catch fires even if it is an error that we expect, however I realised that I hadn't done the additional refactoring I wanted to do last to remove the `returnCode` variable which might make this more clear (updated now).

Do you think it's worth trying to determine which errors are "expected" and which aren't?

Reply:
I'm not sure only errors we expect get here. 
I created a `json_docs/a.json` wiht: 
```json
{
  "_id": "a",
  "_rev": 100
}
```

And called `upload-docs`, knowing that the rev format will throw an error. 
![image](https://user-images.githubusercontent.com/35681649/151914972-ac75216e-85ab-4166-aca9-19f3b45faab9.png)

The error had not been caught by anything else. If I `error(e)`, I get:
![image](https://user-images.githubusercontent.com/35681649/151914917-d694e24f-edb6-493a-bbf6-073a33e7090a.png)

Reply:
Yes I agree, it catches both errors we expect and errors we don't expect. We have a few options to solve this...

1. Output the whole error which is good for unexpected errors, but bad for expected errors because it obscures the message we're trying to send.
2. Output just the message which is good for expected errors but bad for unexpected errors (current solution).
3. Try to distinguish between the two with a special error class or flag. JS doesn't offer a really nice way to do this so I'm not a fan.
4. Revert back somewhat so main returns true/false for success status, and Error for unexpected things.

It sounds like you prefer 1, erring towards too much info rather than hiding it. A fifth option would be a compromise to LOG the entire error, but ERROR the message, so people can see the message really clearly but they also have the more detailed information to help them debug it if the message isn't enough.

What do you think?

Reply:
I'd go for 5, definitely. Anything that helps people debug unexpected errors. 
Thanks for thinking about this!